catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -146,28 +228,39 @@ public Metacard transform(InputStream inputStream)
     @Override
     public Metacard transform(InputStream inputStream, String id)
             throws IOException, CatalogTransformerException {
+        return handleTransform(inputStream, id);
+    }
 
-        String xml = null;
-        XstreamPathValueTracker pathValueTracker = null;
-        xml = IOUtils.toString(inputStream);
+    private Metacard handleTransform(InputStream inputStream, String id)
+            throws CatalogTransformerException {
+        String xml;
+        XstreamPathValueTracker pathValueTracker;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            IOUtils.copy(inputStream, baos);
```

Comment:
Are we guaranteed that this is never going to blow up the heap? What if someone passes in a 10GB XML document. Is this now a DOS attack against the system?